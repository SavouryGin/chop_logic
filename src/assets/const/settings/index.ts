import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
};
