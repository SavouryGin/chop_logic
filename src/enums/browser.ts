export enum Browser {
  Unknown,
  Opera,
  Chrome,
  Safari,
  Firefox,
}
