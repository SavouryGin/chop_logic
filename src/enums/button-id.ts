export enum ButtonID {
  Cancel,
  Navigation,
  ColorTheme,
  Sounds,
  Settings,
  FullScreen,
  Tools,
  Apply,
  Premise,
  Reiteration,
  Replace,
  Delete,
  ImplicationCreation,
  ImplicationDistribution,
  ContradictionRealization,
  ImplicationElimination,
  Assumption,
  NotIntroduction,
  NotElimination,
  AndIntroduction,
  AndElimination,
  OrIntroduction,
  OrElimination,
  ImpliesIntroduction,
  ImpliesElimination,
  EquivIntroduction,
  EquivElimination,
  Shortcut,
  SavePDF,
  ExportXML,
  ImportXML,
  CopyProof,
  PasteProof,
  CutProof,
  Reset,
}
