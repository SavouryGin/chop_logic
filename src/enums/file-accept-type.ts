export enum FileAcceptType {
  Image = 'image/*',
  Video = 'video/*',
  Audio = 'audio/*',
  XML = 'text/xml',
  PDF = 'application/pdf',
  CSV = '.csv',
  Text = 'text/plain',
  Excel = 'application/vnd.ms-excel, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
}
