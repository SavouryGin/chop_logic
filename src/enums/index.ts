export enum Icon {
  Default = 'chop-icon__check',
  Home = 'chop-icon__home',
  Menu = 'chop-icon__menu',
  Enlarge = 'chop-icon__enlarge2',
  Shrink = 'chop-icon__shrink2',
  Settings = 'chop-icon__cog',
  Delete = 'chop-icon__bin2',
  Up = 'chop-icon__circle-up',
  Down = 'chop-icon__circle-down',
  Right = 'chop-icon__circle-right',
  Left = 'chop-icon__circle-left',
  Sound = 'chop-icon__volume-medium',
  NoSound = 'chop-icon__volume-mute2',
  LightMode = 'chop-icon__sun',
  DarkMode = 'chop-icon__contrast',
}

export enum Pages {
  Home,
  NotFound,
  Propositions,
  Predicates,
  TruthTables,
  Syllogisms,
}
