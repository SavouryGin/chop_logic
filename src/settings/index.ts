import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const languageStringOptions = ['en', 'ru'];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  linkedIn: 'https://www.linkedin.com/in/dmitrii-suroviagin/',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
  facebook: 'https://www.facebook.com/dmitrii.suroviagin',
};
