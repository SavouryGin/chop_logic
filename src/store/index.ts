import createSagaMiddleware from 'redux-saga';
import rootReducer from './root-reducer';
import rootSaga from './root-saga';
import { configureStore } from '@reduxjs/toolkit';

const sagaMiddleware = createSagaMiddleware();

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat([sagaMiddleware]),
});

sagaMiddleware.run(rootSaga);
