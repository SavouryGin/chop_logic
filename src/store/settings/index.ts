import { Language } from 'types';
import { PayloadAction, createSlice } from '@reduxjs/toolkit';
import { SettingsFlag, SettingsInitialState } from './interfaces';

export const settingsInitialState: SettingsInitialState = {
  language: 'en',
  flags: {
    isNavigationOpen: false,
    isSidebarOpen: false,
    isDarkMode: false,
    isFullScreen: false,
    isSettingOpen: false,
    isSoundsEnabled: false,
  },
};

export const settingsSlice = createSlice({
  name: 'settings',
  initialState: settingsInitialState,
  reducers: {
    resetSettings: () => {
      return settingsInitialState;
    },

    toggleFlag: (state, action: PayloadAction<SettingsFlag>) => {
      const currentFlagValue = state.flags[action.payload];
      state.flags[action.payload] = !currentFlagValue;
    },

    setLanguage: (state, action: PayloadAction<Language>) => {
      state.language = action.payload;
    },

    setFullScreenFlag: (state, action: PayloadAction<boolean>) => {
      state.flags.isFullScreen = action.payload;
    },

    setUpFlag: (state, action: PayloadAction<{ flag: SettingsFlag; value: boolean }>) => {
      const { flag, value } = action.payload;
      state.flags[flag] = value;
    },
  },
});

export const settingsActions = { ...settingsSlice.actions };
