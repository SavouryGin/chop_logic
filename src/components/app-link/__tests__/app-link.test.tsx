import AppLink from '../index';
import React from 'react';
import renderWithRedux from 'helpers/test-utils/render-with-redux';
import { Icon } from 'enums';
import { combineReducers } from '@reduxjs/toolkit';
import { screen } from '@testing-library/react';
import { settingsInitialState, settingsSlice } from 'store/settings';

describe('AppLink component:', () => {
  const testProps = {
    path: '/test',
    text: 'Test text',
    icon: Icon.Default,
    isNavigation: false,
    className: 'test-class-name',
  };

  const mockedReducer = combineReducers({ settings: settingsSlice.reducer });
  const mockedState = { settings: settingsInitialState };

  beforeEach(() => {
    renderWithRedux(<AppLink {...testProps} />, mockedReducer, mockedState);
  });

  it('renders a link element', () => {
    expect(screen.getByRole('link')).toBeInTheDocument();
  });

  it('the link has href attribute with the correct value', () => {
    const href = `http://localhost${testProps.path}`;
    expect(screen.getByRole('link')).toHaveProperty('href', href);
  });

  it('the link has the passed icon class name', () => {
    expect(screen.getByRole('link')).toHaveClass(testProps.icon);
  });

  it('the link contains the passed text', () => {
    expect(screen.getByRole('link')).toHaveTextContent(testProps.text);
  });

  it('contains all necessary class names', () => {
    expect(screen.getByTestId('app-link')).toHaveClass('app-link', testProps.className);
  });
});
