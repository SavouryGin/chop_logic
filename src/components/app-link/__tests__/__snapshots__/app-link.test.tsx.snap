// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLink component: should match the snapshot 1`] = `
<DocumentFragment>
  <span
    class="app-link test-class-name"
    data-testid="app-link"
  >
    <a
      class="chop-icon__check"
      href="/test"
    >
      Test text
    </a>
  </span>
</DocumentFragment>
`;
